ground/src/test/java/com/google/android/ground/ui/datacollection/DataCollectionFragmentTest.kt

```diff
@@ -13,108 +15,120 @@
+  lateinit var fragment: DataCollectionFragment
 
   @Before
   override fun setUp() {
     super.setUp()
 
-    val activityController = Robolectric.buildActivity(MainActivity::class.java).setup()
-    activity = activityController.get()
+    whenever(
+        surveyRepository.createSubmission(
+          DataCollectionTestData.surveyId,
+          DataCollectionTestData.loiId,
+          DataCollectionTestData.submissionId
+        )
+      )
+      .thenReturn(Single.just(DataCollectionTestData.submission))
   }
 
   @Test
   fun created_submissionIsLoaded() {
     setupFragment()
-    verify(dataCollectionViewModel).loadSubmissionDetails(eq(DataCollectionTestData.args))
+
+    onView(withText(DataCollectionTestData.loiName)).check(matches(isDisplayed()))
   }
 
   @Test
   fun created_submissionIsLoaded_viewPagerAdapterIsSet() {
     setupFragment()
 
-    val viewPager = dataCollectionFragment.view!!.findViewById<ViewPager2>(R.id.pager)
-    assertThat(viewPager).isNotNull()
-    assertThat(viewPager.adapter).isInstanceOf(DataCollectionViewPagerAdapter::class.java)
+    onView(withId(R.id.pager)).check(matches(isDisplayed()))
   }
 
   @Test
-  fun onNextClicked_viewPagerItemIsUpdated() {
+  fun created_submissionIsLoaded_firstTaskIsShown() {
     setupFragment()
 
-    val viewPager = dataCollectionFragment.view!!.findViewById<ViewPager2>(R.id.pager)
-    assertThat(viewPager.currentItem).isEqualTo(0)
-    dataCollectionFragment.view!!
-      .findViewById<Button>(R.id.data_collection_continue_button)
-      .performClick()
-    assertThat(viewPager.currentItem).isEqualTo(1)
+    onView(withText(DataCollectionTestData.task1Name)).check(matches(isDisplayed()))
   }
 
   @Test
-  fun onNextClicked_thenOnBack_viewPagerItemIsUpdated() {
+  fun onContinueClicked_noUserInput_toastIsShown() {
     setupFragment()
 
-    val viewPager = dataCollectionFragment.view!!.findViewById<ViewPager2>(R.id.pager)
-    assertThat(viewPager.currentItem).isEqualTo(0)
-    dataCollectionFragment.view!!
-      .findViewById<Button>(R.id.data_collection_continue_button)
-      .performClick()
-    assertThat(viewPager.currentItem).isEqualTo(1)
-    assertThat(dataCollectionFragment.onBack()).isTrue()
-    assertThat(viewPager.currentItem).isEqualTo(0)
+    onView(withId(R.id.data_collection_continue_button)).perform(click())
+
+    assertThat(ShadowToast.getTextOfLatestToast()).isEqualTo("This field is required")
   }
 
   @Test
-  fun onBack_firstViewPagerItem_returnsFalse() {
+  fun onContinueClicked_newTaskIsShown() {
     setupFragment()
+    onView(allOf(withId(R.id.user_response_text), isDisplayed())).perform(typeText("user input"))
+
+    onView(withId(R.id.data_collection_continue_button)).perform(click())
 
-    assertThat(dataCollectionFragment.onBack()).isFalse()
+    onView(withText(DataCollectionTestData.task1Name)).check(matches(not(isDisplayed())))
+    onView(withText(DataCollectionTestData.task2Name)).check(matches(isDisplayed()))
```

Comment:
Similar to the above case, can we assert that no toast was shown in this case?

Reply:
Done!